`test_etag_sync` fails on khal 0.10.4

`test_etag_sync` fails on khal 0.10.4 on openSUSE Tumbleweed aarch64 and armv7.

```
[   82s] =================================== FAILURES ===================================
[   82s] ________________________________ test_etag_sync ________________________________
[   82s] 
[   82s] tmpdir = local('/tmp/pytest-of-abuild/pytest-1/test_etag_sync0')
[   82s] 
[   82s]     def test_etag_sync(tmpdir):
[   82s]         fpath = os.path.join(str(tmpdir), 'foo')
[   82s]     
[   82s]         file_ = open(fpath, 'w')
[   82s]         file_.write('foo')
[   82s]         file_.close()
[   82s]         os.sync()
[   82s]         old_etag = vdir.get_etag_from_file(fpath)
[   82s]     
[   82s]         file_ = open(fpath, 'w')
[   82s]         file_.write('foo')
[   82s]         file_.close()
[   82s]     
[   82s]         new_etag = vdir.get_etag_from_file(fpath)
[   82s]     
[   82s] >       assert old_etag != new_etag
[   82s] E       AssertionError: assert '1642549344080000000.000000000' != '1642549344080000000.000000000'
[   82s] 
[   82s] tests/vdir_test.py:68: AssertionError
```

--- a/tests/vdir_test.py
+++ b/tests/vdir_test.py
@@ -66,7 +66,12 @@ def test_etag_sync(tmpdir):
 
     new_etag = vdir.get_etag_from_file(fpath)
 
-    assert old_etag != new_etag
+    try:
+        assert old_etag != new_etag
+    except AssertionError:
+        pytest.xfail(
+            "Do we need to sleep?"
+        )
 
 
 def test_etag_sleep(tmpdir, sleep_time):

--- a/tests/vdir_test.py
+++ b/tests/vdir_test.py
@@ -27,8 +27,10 @@ import pytest
 from khal.khalendar import vdir
 
 
-@pytest.mark.xfail
-def test_etag(tmpdir):
+def test_etag(tmpdir, sleep_time):
+    if sleep_time > 0.01:
+        pytest.xfail("This environment requires sleeping")
+
     fpath = os.path.join(str(tmpdir), 'foo')
 
     file_ = open(fpath, 'w')
@@ -43,15 +45,13 @@ def test_etag(tmpdir):
 
     new_etag = vdir.get_etag_from_file(fpath)
 
-    try:
-        assert old_etag != new_etag
-    except AssertionError:
-        pytest.xfail(
-            "Do we need to sleep?"
-        )
+    assert old_etag != new_etag
 
 
-def test_etag_sync(tmpdir):
+def test_etag_sync(tmpdir, sleep_time):
+    if sleep_time > 0.01:
+        pytest.xfail("This environment requires sleeping")
+
     fpath = os.path.join(str(tmpdir), 'foo')
 
     file_ = open(fpath, 'w')
@@ -66,12 +66,7 @@ def test_etag_sync(tmpdir):
 
     new_etag = vdir.get_etag_from_file(fpath)
 
-    try:
-        assert old_etag != new_etag
-    except AssertionError:
-        pytest.xfail(
-            "Do we need to sleep?"
-        )
+    assert old_etag != new_etag
 
 
 def test_etag_sleep(tmpdir, sleep_time):

